app/src/test/kotlin/compiler/intermediate_form/DefaultDefaultFunctionDetailsGeneratorTest.kt

```diff
@@ -0,0 +1,348 @@
+package compiler.intermediate_form
+
```

Comment:
We don't have tests for `genPrologue`/`genEpilogue`, but these would probably be ugly tests.

Reply:
I can't see any way to test it without repeating exactly the same code in these tests as it is in the `genPrologue/Epilogue`, so it makes no sense to me. Maybe we could check if the order of pushing/popping items to stack is consistent between prologue and epilogue, but it is still only a fraction of the responsibility of these methods. The only method of testing it that comes to my mind are some integration tests later, when we'll have full translation to assembler - we'd compile it, call, and assert there were no runtime errors in asm code.

Reply:
Sounds reasonable.